api/build.gradle

```diff
@@ -9,17 +9,42 @@ plugins {
 
 description = 'gRPC: API'
 
+sourceSets {
+    context {
+        java {
+            srcDir "${projectDir}/context/src/main/java"
+        }
+    }
+    main {
+        output.classesDirs.from(sourceSets.context.output.classesDirs)
+    }
+    test {
+        java {
+            srcDir "${projectDir}/context/src/test/java"
+        }
+    }
+}
+
+compileContextJava {
+    sourceCompatibility = 1.7
+    targetCompatibility = 1.7
+}
+
+
 dependencies {
-    api project(':grpc-context'),
-            libraries.jsr305,
+    compileOnly sourceSets.context.output
+    testFixturesApi libraries.truth
+    // Explicitly choose the guava version to stay Java 7-compatible.
```

Comment:
This may not be necessary any more, and looks like it isn't doing anything. That existed because source/target compatibility applied to tests as well. With the tests here, like this, the tests are not Java 7 compatible. And the version of gradle being used here is actually newer, because `testFixturesImplementation libraries.guava,` is depending on a newer gradle.